src/GafferUIBindings/PlugAdderBinding.cpp

```diff
@@ -92,57 +92,56 @@ struct PlugMenuSlotCaller
 
 };
 
-struct PlugAdderWrapper : public PlugAdder, IECorePython::Wrapper< PlugAdder >
+struct PlugAdderWrapper : public GadgetWrapper<PlugAdder>
 {
 
-	PlugAdderWrapper(PyObject *self, StandardNodeGadget::Edge edge ) : PlugAdder( edge ), IECorePython::Wrapper<PlugAdder>( self, this )
+	PlugAdderWrapper(PyObject *self, StandardNodeGadget::Edge edge )
+		: GadgetWrapper<PlugAdder>( self, edge )
 	{
 	}
 
-	bool acceptsPlug( const Gaffer::Plug *connectionEndPoint ) const
+	virtual bool acceptsPlug( const Gaffer::Plug *connectionEndPoint ) const
 	{
-		IECorePython::ScopedGILLock gilLock;
-		override o = this->get_override( "acceptsPlug" );
-		if( o )
+		if( this->isSubclassed() )
 		{
-			try
+			IECorePython::ScopedGILLock gilLock;
+			boost::python::object f = this->methodOverride( "acceptsPlug" );
+			if( f )
 			{
-				return o( Gaffer::PlugPtr( const_cast<Gaffer::Plug *>( connectionEndPoint ) ) );
+				try
+				{
+					return f( Gaffer::PlugPtr( const_cast<Gaffer::Plug *>( connectionEndPoint ) ) );
+				}
+				catch( const error_already_set &e )
+				{
+					translatePythonException();
+				}
 			}
-			catch ( error_already_set )
-			{
-				PyErr_Print();
-			}
-		}
-		else
-		{
-			// TODO: replace this with proper logging?
-			printf( "PlugAdder: acceptsPlug() python method not defined" );
 		}
-
-		return false;
-
+		throw IECore::Exception( "No acceptsPlug method defined in Python." );
 	}
 
 	virtual void addPlug( Gaffer::Plug *connectionEndPoint )
 	{
-		IECorePython::ScopedGILLock gilLock;
-		override o = this->get_override( "addPlug" );
-		if( o )
+		if( this->isSubclassed() )
 		{
-			try
-			{
-				o( Gaffer::PlugPtr( const_cast<Gaffer::Plug *>( connectionEndPoint ) ) );
-			}
-			catch ( error_already_set )
+			IECorePython::ScopedGILLock gilLock;
+			boost::python::object f = this->methodOverride( "addPlug" );
+			if( f )
 			{
-				PyErr_Print();
+				try
+				{
+					f( Gaffer::PlugPtr( const_cast<Gaffer::Plug *>( connectionEndPoint ) ) );
+				}
+				catch( const error_already_set &e )
+				{
+					translatePythonException();
+				}
 			}
 		}
 		else
 		{
-			// TODO: replace this with proper logging?
-			printf( "PlugAdder: addPlug() python method not defined" );
+			throw IECore::Exception( "No addPlug method defined in Python." );
```

Comment:
I think this needs to come out of the `else` block so the structure is the same as for acceptPlug above? Otherwise we don't throw if the PlugAdder has been subclasses but the override hasn't been implemented. We'll need a return statement below the call to `f()` as well of course.